static/app/components/workflowEngine/layout/edit.tsx

```diff
@@ -0,0 +1,118 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+import type {LocationDescriptor} from 'history';
+
+import {Breadcrumbs} from 'sentry/components/breadcrumbs';
+import ButtonBar from 'sentry/components/buttonBar';
+import * as Layout from 'sentry/components/layouts/thirds';
+import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
+import {space} from 'sentry/styles/space';
+
+export interface WorkflowEngineEditLayoutProps {
+  /**
+   * The main content for this page
+   */
+  children: React.ReactNode;
+  /**
+   * Breadcrumb info for the parent page
+   */
+  parent: {
+    /**
+     * Link to the parent page, displayed as a breadcrumb
+     */
+    href: LocationDescriptor;
+    /**
+     * Title of the parent page, displayed as a breadcrumb
+     */
+    title: React.ReactNode;
+  };
+  /**
+   * Action buttons to display in the page header
+   */
+  actions?: React.ReactNode;
+
+  /**
+   * The main page title, for example "Automations" or "Rules"
+   */
+  title?: string;
+}
+
+/**
+ * Precomposed full-width layout for Automations / Rules edit pages.
+ * The `children` are expected to include `<EditLayout.Chart>` and `<EditLayout.Panel>` components.
+ */
+function EditLayout({parent, children, actions, title}: WorkflowEngineEditLayoutProps) {
+  return (
+    <Fragment>
+      <SentryDocumentTitle title={title} />
+      <Layout.Page>
+        <StyledHeader>
+          <Layout.HeaderContent>
+            <Breadcrumbs
+              crumbs={[
+                {
+                  label: parent.title,
+                  to: parent.href,
+                },
+                {
+                  label: title,
+                },
+              ]}
+            />
+            <Layout.Title>{title}</Layout.Title>
+          </Layout.HeaderContent>
+          {actions ? (
+            <Layout.HeaderActions>
+              <ButtonBar merged={false} gap={1}>
+                {actions}
+              </ButtonBar>
+            </Layout.HeaderActions>
+          ) : null}
+        </StyledHeader>
+        <Body>{children}</Body>
+      </Layout.Page>
+    </Fragment>
+  );
+}
+
+const StyledHeader = styled(Layout.Header)`
+  background: ${p => p.theme.background};
+`;
+
+const Body = styled('div')`
+  display: flex;
+  flex-direction: column;
+  gap: ${space(3)};
```

Comment:
Thoughts on using a CSS grid for the layout rather than flexbox? that gives us a little more flexibility on column / rows and general layout positioning. Idk if it's hugely important, but if we want to try out css grids, now is probably the best time. If we don't use grids, should we use the `Flex` component for these props instead?

Reply:
Flex seems most appropriate here since we have a fairly straightforward vertical layout in a single column, where each row might have a different height depending on it's content. I'd reach for grid if we had multiple columns/rows or if the row height was dictated by the containing item rather than the children.